vllm_spyre/v1/worker/spyre_worker.py

```diff
@@ -401,11 +401,24 @@ def _warmup_spyre_dynamic_size(self, special_token_ids):
             logger.info("update_lazyhandle() done (duration: %.3fs)",
                         ul_stop_time - ul_start_time)
 
+        # get the number or pages from the actual Spyre card after the warmup
+        # and set it accordingly in the model runner and the kv cache size
+        n_blocks_spyre = self.get_num_blocks_from_compiler_mock()
+        model_runner._set_free_blocks(num_blocks=n_blocks_spyre)
+        model_runner.model.model._set_past_key_value_states(
+            num_blocks=n_blocks_spyre)
```

Comment:
I realised all these changes only applies to `sendnn_decoder` backend and we still need the old logic for the other backends.

Reply:
Maybe everything works by renaming function `get_num_blocks_from_compiler_mock` to `get_num_blocks` and either return `max_batch_size * max_model_len // self.BLOCK_SIZE` if cpu backend or call the torch_sendnn function if `torch_sendnn` backend